scripts/qemucommand.py

```diff
@@ -129,10 +144,15 @@ def command_line(self):
             netuser += ',hostfwd=tcp:0.0.0.0:2159-:2159'
         cmdline = [
             "qemu-system-x86_64",
-            "-bios", self.bios
         ]
+        if self.enable_u_boot:
+            cmdline += ["-bios", self.bios]
+        else:
+            cmdline += ["-device", "virtio-net-pci"]
+            cmdline += ["-kernel", self.kernel]
```

Comment:
Interesting that we have to supply the kernel. Why do we need to here if we don't have to in the OSTree/u-boot case?

Reply:
When we use u-boot, we specify it using -bios option in QEMU, so (u-boot.rom) is interpreted by QEMU as a kind of BIOS, and is responsible to load the kernel. In the case of QEMU, this is just a path to the kernel binary file i.e. 
./tmp/deploy/image/qemux86-64/bzImage
A relative path to the kernel image is specified in the patch to u-boot
https://github.com/advancedtelematic/meta-updater-qemux86-64/blob/b1d134cbb5a47e5008db4a65510b4c102f636b4d/recipes-bsp/u-boot/files/0001-Set-up-environment-for-OSTree-integration.patch#L41
To check it out we can simple remove\rename this file
./tmp/deploy/images/qemux86-64/bzImage
which is just a link to the actual kernel binary file, and the run-qemu-ota won't work because u-boot won't be able to find kernel image.